src/bootstrap/config.rs

```diff
@@ -878,8 +878,10 @@ impl Default for StringOrBool {
 #[derive(Clone, Debug, Deserialize, PartialEq, Eq)]
 #[serde(untagged)]
 pub enum RustOptimize {
-    #[serde(deserialize_with = "deserialize_and_validate_opt_level")]
+    #[serde(deserialize_with = "deserialize_and_validate_opt_level_string")]
```

Comment:
I didn't use `StringOrInt` here because it will introduce a lifetime into the RustOptimize and the whole Config.

I also tried to change the StrubfOrInt to use String instead of `&'a str`, but it didn't work. Because we use the match patterns when deserializing the DebuginfoLevel. We can not match it with `StringOrInt::String(String::from_str("none"))`.

Do you have any thoughts about how to improve it? Or this approach is OK?